app/controllers/user_registrations_controller.rb

```diff
@@ -5,14 +5,13 @@ class UserRegistrationsController < Spree::UserRegistrationsController
   def create
     @user = build_resource(params[:spree_user])
     if resource.save
-      set_flash_message(:success, :signed_up)
-      sign_in(:spree_user, @user)
       session[:spree_user_signup] = true
       associate_user
 
       respond_to do |format|
         format.html do
-          sign_in_and_redirect(:spree_user, @user)
+          set_flash_message(:success, :signed_up_but_unconfirmed)
+          redirect_to after_sign_in_path_for(@user)
         end
         format.js do
           render json: { email: @user.email }
```

Comment:
I don't know why this file isn't marked as a `_decorator`. Something to do in a separate PR.